metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,147 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codeactions.CodeAction
+import scala.meta.internal.metals.codeactions.MillifyScalaCliDependencyCodeAction._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+import scala.meta.tokens.Token
+import scala.meta.tokens.Tokens
+
+import org.eclipse.{lsp4j => l}
+
+class ConvertCommentCodeAction(buffers: Buffers) extends CodeAction {
+
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+    val couldBeScalaCli = path.isScalaScript || path.isScala
+
+    (for {
+      content <- buffers.get(path)
+      tokens <- tokenizeIfNotRangeSelection(range, content)
+      contentLines = content.split("\\r?(\\n|\\r)").toVector
+      codeAction <- createActionIfPossible(
+        path,
+        tokens,
+        range,
+        isSingleLineComment(contentLines, couldBeScalaCli),
+      )
+    } yield codeAction).toList
+  }
+
+  private def tokenizeIfNotRangeSelection(range: l.Range, content: String) = {
+    Option
+      .when(range.getStart == range.getEnd)(
+        Trees.defaultTokenizerDialect(content).tokenize
+      )
+      .flatMap(_.toOption)
+  }
+
+  private def isSingleLineComment(
+      contentLines: Vector[String],
+      couldBeScalaCli: Boolean,
+  )(
+      t: Token
+  ): Boolean = t match {
+    case tc: Token.Comment =>
+      val currentLine = contentLines(t.pos.startLine)
+      val isLikelyScalaCliDirective =
```

Comment:
Oh, good thinking! :+1: Would be nice to add a simple unit test for this case. :slightly_smiling_face:

Reply:
Actually, I added it because another test was failing because the reported actions didn't match anymore (`MillifyScalaCliDependencies` or something like that). 

Should we add another test for that in our test-suite?

Reply:
Ah, I see. :+1: Honestly, I think it's still a really good idea. Converting scala-cli directives is probably not something that would be typically desired (could it even break them?), so I feel it makes perfect sense to have this adjustment in place. In which case I'd personally add at least a single unit test related to that for this code action, but it's just my 2 cents. :slightly_smiling_face: